codes/ruby/chapter_backtracking/preorder_traversal_i_compact.rb

```diff
@@ -0,0 +1,33 @@
+=begin
+File: preorder_traversal_i_compact.rb
+Created Time: 2024-04-07
+Author: hoangtien2k3 (hoangtien2k3qx1@gmail.com)
+=end
+
+require_relative 'utils'
```

Comment:
Ruby cannot load the files in /utils with this `require_relative`

```suggestion
require_relative '../utils/tree_node'
require_relative '../utils/print_util'
```